crates/router/src/core/admin.rs

```diff
@@ -3716,54 +3703,85 @@ pub async fn list_business_profile(
         .await
         .to_not_found_response(errors::ApiErrorResponse::MerchantAccountNotFound)?;
     let profiles = db
-        .list_business_profile_by_merchant_id(&(&state).into(), &key_store, &merchant_id)
+        .list_profile_by_merchant_id(&(&state).into(), &key_store, &merchant_id)
         .await
         .to_not_found_response(errors::ApiErrorResponse::InternalServerError)?
         .clone();
     let profiles = core_utils::filter_objects_based_on_profile_id_list(profile_id_list, profiles);
     let mut business_profiles = Vec::new();
     for profile in profiles {
-        let business_profile =
-            api_models::admin::BusinessProfileResponse::foreign_try_from(profile)
-                .change_context(errors::ApiErrorResponse::InternalServerError)
-                .attach_printable("Failed to parse business profile details")?;
+        let business_profile = api_models::admin::ProfileResponse::foreign_try_from(profile)
+            .change_context(errors::ApiErrorResponse::InternalServerError)
+            .attach_printable("Failed to parse business profile details")?;
+        business_profiles.push(business_profile);
+    }
+
+    Ok(service_api::ApplicationResponse::Json(business_profiles))
+}
+
+#[cfg(all(feature = "olap", feature = "v2"))]
+pub async fn list_profile(
+    state: SessionState,
+    merchant_id: id_type::MerchantId,
+    profile_id_list: Option<Vec<id_type::ProfileId>>,
+) -> RouterResponse<Vec<api_models::admin::ProfileResponse>> {
+    let db = state.store.as_ref();
+    let key_store = db
+        .get_merchant_key_store_by_merchant_id(
+            &(&state).into(),
+            &merchant_id,
+            &db.get_master_key().to_vec().into(),
+        )
+        .await
+        .to_not_found_response(errors::ApiErrorResponse::MerchantAccountNotFound)?;
+    let profiles = db
+        .list_profile_by_merchant_id(&(&state).into(), &key_store, &merchant_id)
+        .await
+        .to_not_found_response(errors::ApiErrorResponse::InternalServerError)?
+        .clone();
+    let profiles = core_utils::filter_objects_based_on_profile_id_list(profile_id_list, profiles);
+    let mut business_profiles = Vec::new();
+    for profile in profiles {
+        let business_profile = api_models::admin::ProfileResponse::foreign_try_from(profile)
```

Comment:
Should we consider calling this as `profile_response`, and renaming `business_profiles` to something else?

Reply:
we are taking the variable name changes in upcoming pull request